src/lib/connect/index.ts

```diff
@@ -104,6 +104,19 @@ function connect(
 		opts.clientId = opts.query.clientId
 	}
 
+	if (isBrowser || opts.unixSocket) {
+		opts.socks = undefined
+	}
+
+	if (
+		!isBrowser &&
+		!opts.unixSocket &&
+		opts.socks === undefined &&
+		typeof process !== 'undefined'
+	) {
+		opts.socks = process.env['MQTTJS_SOCKS_PROXY']
+	}
```

Comment:
why? Could you remove the env var here?

Reply:
I added the environment variable in order to provide an easy way to route unmodified applications that use mqtt.js through a proxy. In or case the culprit is Node-RED.

It would be definitely preferable if these had support for using a proxy themselves, but configuring a proxy is one of those networking options where it is not uncommon to offer configuration through the environment.